asciidoctor-maven-plugin/src/main/java/org/asciidoctor/maven/process/CopyResourcesProcessor.java

```diff
@@ -116,14 +116,15 @@ private List<Resource> prepareResources(File sourceDirectory, AsciidoctorMojo co
      *
      * @param resources       Collection of {@link Resource} defining what resources to {@code outputDirectory}.
      * @param outputDirectory Directory where to copy resources.
+     * @param configuration   Project configuration
      */
-    private void copyResources(List<Resource> resources, File outputDirectory) {
+    private void copyResources(List<Resource> resources, File outputDirectory, AsciidoctorMojo configuration) {
 
         resources.stream()
-            .filter(resource -> new File(resource.getDirectory()).exists())
+            .filter(resource -> configuration.getBaseDir().exists())
```

Comment:
I think that what we want here is `getProjectDirectory()`, thinking about it is not as simple as I initially though.

There's the "project's baseDir", that's the path in which the pom resides (this is set always by maven using this [annotation here](https://github.com/asciidoctor/asciidoctor-maven-plugin/blob/5077f0425b9cae3dff6f37039fcdb6e1a5c6caf9/asciidoctor-maven-plugin/src/main/java/org/asciidoctor/maven/AsciidoctorMojo.java#L73)). And also "Asciidoctor's baseDir" which is an option to set the root where Asciidoctor searches for files (likes includes).
The latter (which is being used in this code) can be null, and I don't think we should deal with it. It's my assumption that in `<resources>` the users will want to put either an absolute path or a relative path from the pom's location.

That means we should check if the path in the `<resource>` is absolute or not.
If it's absolute, we do the same we do now, if it's not, we need to compose that with `configuration.getProjectDirectory()`. On way to do it would be adding creating a new Resource instance before the filter using `map()`.
```java
            .map(resource -> {
                final File candidate = new File(resource.getDirectory());
                final File sourceDir = candidate.isAbsolute() ? candidate : new File(configuration.getProjectDirectory(), resource.getDirectory());

                final Resource sanitizedResource = new Resource();
                sanitizedResource.setDirectory(sourceDir.getAbsolutePath());
                sanitizedResource.setTargetPath(resource.getTargetPath());
                sanitizedResource.setIncludes(resource.getIncludes());
                sanitizedResource.setExcludes(resource.getExcludes());
                return sanitizedResource;
            })
```

I haven't tested myself, but I think it should work :thinking: so all these work
```
    <resource>
        <directory>src/docs/asciidoc/images/</directory>
        <targetPath>images/</targetPath>
        <includes>
            <include>*.png</include>
        </includes>
    </resource>
```

```
    <resource>
        <directory>${project.basedir}/src/docs/asciidoc/images/</directory>
        <targetPath>images/</targetPath>
        <includes>
            <include>*.png</include>
        </includes>
    </resource>
```


Btw, if a user inputs `<directory>/src/docs/asciidoc/images/</directory>`, to me that's a configuration error and it should be considered an absolute path. We shouldn't try to guess and interpret `/`.

Reply:
Oh... I see. I'll try)

Reply:
Tried, seems working

Reply:
Thanks, I am quite busy, so I'll check first thing dueing the weekend.